semanticcpg/src/main/scala/io/shiftleft/semanticcpg/passes/BindingMethodOverridesPass.scala

```diff
@@ -17,14 +19,18 @@ class BindingMethodOverridesPass(cpg: Cpg) extends CpgPass(cpg) {
       .toIterator()
       .foreach(binding => {
         val typeDecl: Option[nodes.TypeDecl] = binding.start.bindingTypeDecl.headOption()
-        assert(typeDecl.isDefined,
-               "No binding typeDecl found in BindingMethodOverridesPass: " + method.name + " " + method.signature)
-        val neverOverriddenFlag = isMethodNeverOverridden(typeDecl.get, method.name, method.signature)
-        diff.addNodeProperty(
-          binding,
-          NodeKeys.IS_METHOD_NEVER_OVERRIDDEN.name,
-          neverOverriddenFlag.asInstanceOf[AnyRef]
-        )
+        if (typeDecl.isDefined) {
+          val neverOverriddenFlag = isMethodNeverOverridden(typeDecl.get, method.name, method.signature)
+          diff.addNodeProperty(
+            binding,
+            NodeKeys.IS_METHOD_NEVER_OVERRIDDEN.name,
+            neverOverriddenFlag.asInstanceOf[AnyRef]
+          )
+        } else {
+          logger.error(
+            "No binding typeDecl found in BindingMethodOverridesPass: " + method.name + " " + method.signature)
+          true
```

Comment:
With `foreach` don't need the return value do you?

Reply:
@Ferada Correct, thanks